### PR TITLE
add initial monitoring support to cloud run jobs

### PR DESCRIPTION
**Pull request by shankiyani**

### CODE REVIEW

File: terraform/modules/alerts/alerts.tf [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  hour   = 60 * local.minute
+  day    = 24 * local.hour
+
+  forward_progress_job_indicators = merge(
+    {
+      # artifacts job runs every 15m, alert after 3 failures + buffer
+      "gma-artifacts" = { metric = "completed_execution_count", window = 45 * local.minute + 5 * local.minute },
+
+      # review status job runs every 4h, alert after 2 failures + buffer
+      "commit-review-status-job" = { metric = "completed_execution_count", window = 8 * local.hour + 10 * local.minute },
```
[the remark this answers is not in the document]
Comment (shankiyani) [reply terraform/modules/alerts/alerts.tf:30]: Given the cron syntax we cant directly correlate. Currently you can use the variable `forward_progress_job_indicators` to override the predefined ones so we actually have this support built in.

**Review by lock14 — APPROVED** [review]

File: terraform/variables.tf [hunk 2]
```diff
@@ -429,3 +429,14 @@ variable "default_log_bucket_configuration" {
...
 
+variable "alerts" {
+  description = "The configuration block for alerts and notifications"
+  type = object({
+    enabled = bool
+    email   = string
+  })
+  default = {
+    enabled = false
+    email   = "team@email.com"
```
Comment (verbanicm) [inline terraform/variables.tf:440]: Can we make this an empty string?

File: terraform/modules/alerts/variables.tf [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  type        = string
+}
+
+variable "alert_notification_channel_non_paging" {
+  description = "Non-paging notification channels"
+  type        = map(any)
+  default = {
+    email = {
+      labels = {
+        email_address = "team@email.com"
```
Comment (verbanicm) [inline terraform/modules/alerts/variables.tf:26]: Empty string

File: terraform/modules/alerts/terraform.tf [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+terraform {
+  required_version = ">= 1.7"
+
+  required_providers {
+    google = {
+      version = ">= 5.19"
```
Comment (verbanicm) [inline terraform/modules/alerts/terraform.tf:20]: Should this use `~>` instead?

**Review by verbanicm — APPROVED** [review]

**Pull request merged**